gui/src/3D/controls/touchToggleButton3D.ts

```diff
@@ -0,0 +1,79 @@
+import { AbstractMesh } from "babylonjs/Meshes/abstractMesh";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { Observable } from "babylonjs/Misc/observable";
+import { Scene } from "babylonjs/scene";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Vector3 } from "babylonjs/Maths/math.vector";
+import { Vector3WithInfo } from "../vector3WithInfo";
+
+import { TouchButton3D, ButtonState } from "./touchButton3D";
+
+/**
+ * Class used as base class for touch-enabled toggleable buttons
+ */
+export class TouchToggleButton3D extends TouchButton3D {
+    private _isPressed = false;
+
+    /** Callback used to start toggle on animation */
+    public toggleOnAnimation: () => void;

```

Comment:
are these callbacks needed? why not use the observables? We are trying to avoid direct callbacks and are always pushing towards using observables instead

Reply:
Oh. I did this since Control3D has callbacks for the animations.

Reply:
I actually was never sure why there were both in Control3D, since they were both called at the same point. But if callbacks are deprecated, then it makes more sense.

Reply:
Not really deprecated but not needed. Unless i am missing some inheritance issue?